dawn/renderer/utils/lcm_ws_bridge.js

```diff
@@ -0,0 +1,148 @@
+
+  This library imitates the LCM APIs, and fulfills requests by forwarding them
+  to a websocket server.
+
+  PUBLIC API:
+
+    new LCM(websocket_url)
+    on_ready(callback)
+    subscribe(channel, msg_type, callback)
+    unsubscribe(subscription_id)
+    publish(channel, msg)
+
+  EXAMPLE CODE:
+
+    let l = new LCM("ws://localhost:8000");
+
+    l.on_ready(function() {
+      let sub = l.subscribe("sprocket/health", "health", function(msg) {
+        alert(msg.header.seq);
+      });
+
+      window.setTimeout(function() {
+        l.unsubscribe(sub);
+      }, 1000);
+
+      l.publish("/forest/cmd", {
+        __type__: "forest_cmd",
+        header: {
+          __type__: "header",
+          seq: 5,
+          time: 0
+        },
+        lights: [[true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false]],
+        servos: [0, 0, 0, 0, 0, 0, 0, 0]
+      });
+    });
+*/
+import WebSocket from 'ws';
+
+
+const CircularJSON = require('circular-json');
+
+class LCM {
+  constructor(wsUri) {
+    this.ws = new WebSocket(wsUri);
+    this.ws.onmessage = (evt) => {
+      this.delegate(CircularJSON.parse(evt.data));
+    };
+    this.callbacks = {}; // indexed by subscription id
+    this.on_ready = this.on_ready.bind(this);
+    this.on_close = this.on_close.bind(this);
+    this.ws_send = this.ws_send.bind(this);
+    this.delegate = this.delegate.bind(this);
+    this.generate_uuid = this.generate_uuid.bind(this);
+    this.subscribe = this.subscribe.bind(this);
+    this.unsubscribe = this.unsubscribe.bind(this);
+    this.publish = this.publish.bind(this);
+  }
+
+  on_ready(cb) { // eslint-disable-line camelcase
+    // Call the callback once this LCM instance is ready to receive commands
+    this.ws.onopen = () => { cb(this); };
+  }
+
+  on_close(cb) { // eslint-disable-line camelcase
+    // Call the callback once this LCM instance is ready to receive commands
+    this.ws.onclose = () => { cb(this); };
+  }
+
+  ws_send(type, data) { // eslint-disable-line camelcase
+    // Internal convenience method for sending data over the websocket
+    try {
+      this.ws.send(CircularJSON.stringify({ type, data }));
+    } catch (err) {
+      console.log('*');
+    }
+  }
+
+  delegate(request) {
+    // Internal method that delegates data received to the appropriate handler
+    let callback;
+    switch (request.type) {
+      case 'packet':
+        callback = this.callbacks[request.data.subscription_id];
+        if (callback !== undefined) {
+          callback(request.data.msg);
+        }
+        break;
+      default:
+        throw 'Invalid request!'; // eslint-disable-line no-throw-literal
+    }
+  }
+
+  generate_uuid() { // eslint-disable-line
```

Comment:
Is there a reason why we don't just use [node-uuid](https://github.com/kelektiv/node-uuid) here?